### PR TITLE
Compat for bokeh 0.12.10

### LINKED ISSUE #2843: dask.diagnostics.visualize incompatible with bokeh 0.12.10 (latest)
I am getting this stack trace when calling `dask.diagnostics.visualize` with `dask==0.15.4` and `bokeh==0.12.10`:

```
Traceback (most recent call last):
  File "scripts/compress_volumes_dask.py", line 173, in <module>
    run_pipeline()
  File "scripts/compress_volumes_dask.py", line 170, in run_pipeline
    visualize([prof, rprof, cprof])
  File "/Users/jcwork/.pyenv/versions/capstone/lib/python3.5/site-packages/dask/diagnostics/profile_visualize.py", line 171, in visualize
    from bokeh.io import _state
ImportError: cannot import name '_state'
```

It appears that bokeh.io no longer exports a _state variable. I'm not familiar with any of the code here -- possibly this now wants to be fetched from `bokeh.io.state.curstate()`?

For now, installing the previous version of bokeh, 0.12.9, fixes the problem.

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes #2843

### CODE REVIEW

**Comment by jcrist** [comment]
Looks good to me. Thanks @TomAugspurger.

File: dask/diagnostics/profile_visualize.py [hunk 1]
```diff
@@ -168,9 +169,16 @@ def visualize(profilers, file_path=None, show=True, save=True, **kwargs):
     The completed bokeh plot object.
     """
     bp = import_required('bokeh.plotting', _BOKEH_MISSING_MSG)
-    from bokeh.io import _state
+    import bokeh
 
-    if not _state._notebook:
+    if bokeh.__version__ >= LooseVersion("0.12.10"):
```
Comment (jcrist) [inline dask/diagnostics/profile_visualize.py]: You want `LooseVersion(bokeh.__version__) >= "0.12.10"`.

Comment (TomAugspurger) [reply dask/diagnostics/profile_visualize.py]: Whoops... thanks.

**Comment by bryevdv** [comment]
`curstate` has existed since version `0.11` two years ago. You should just update to use that, always. Versions older than `0.11` are at most of historical interest.

**Pull request merged**